pandas/tests/dtypes/test_inference.py

```diff
@@ -892,25 +872,25 @@ def test_infer_dtype_period(self):
         arr = np.array([pd.Period("2011-01", freq="D"), pd.Period("2011-02", freq="M")])
         assert lib.infer_dtype(arr, skipna=True) == "period"
 
-        # starts with nan
```

Comment:
IIRC this comment was related to needing a test specifically for the NaN coming before the period.  can you add a comment in the appropriate place to that effect